libnmstate/state.py

```diff
@@ -32,6 +32,72 @@
 from libnmstate.schema import Interface
 from libnmstate.schema import InterfaceState
 from libnmstate.schema import InterfaceType
+from libnmstate.schema import Route
+
+
+class RouteEntry(object):
+
+    # Mapping dict key name to RouteEntry property name
+    KEY_PROPERTY_MAP = {
+        Route.STATE: Route.STATE.replace('-', '_'),
+        Route.TABLE_ID: Route.TABLE_ID.replace('-', '_'),
+        Route.METRIC: Route.METRIC.replace('-', '_'),
+        Route.NEXT_HOP_ADDRESS: Route.NEXT_HOP_ADDRESS.replace('-', '_'),
+        Route.DESTINATION: Route.DESTINATION.replace('-', '_'),
+        Route.NEXT_HOP_INTERFACE: Route.NEXT_HOP_INTERFACE.replace('-', '_'),
+    }
+
+    def __init__(self, route):
+        self.table_id = route.get(Route.TABLE_ID)
+        self.state = route.get(Route.STATE)
+        self.metric = route.get(Route.METRIC)
+        self.destination = route.get(Route.DESTINATION)
+        self.next_hop_address = route.get(Route.NEXT_HOP_ADDRESS)
+        self.next_hop_interface = route.get(Route.NEXT_HOP_INTERFACE)
+        self.complement_defaults()
+
+    def complement_defaults(self):
+        if self.state != Route.STATE_ABSENT:
+            if self.table_id is None:
+                self.table_id = Route.USE_DEFAULT_ROUTE_TABLE
+            if self.metric is None:
+                self.metric = Route.USE_DEFAULT_METRIC
+            if self.next_hop_address is None:
+                self.next_hop_address = ''
+
+    def __hash__(self):
+        return hash(
+            tuple(getattr(self, prop)
+                  for prop in six.viewvalues(RouteEntry.KEY_PROPERTY_MAP)))
+
+    def to_dict(self):
+        route_entry = {}
+        for key, prop in six.viewitems(RouteEntry.KEY_PROPERTY_MAP):
+            if getattr(self, prop) is not None:
+                route_entry[key] = getattr(self, prop)
+        return route_entry
+
+    def __eq__(self, other):
+        return self is other or self.to_dict() == other.to_dict()
+
+    def can_remove(self, other):
```

Comment:
You previous version of `match` was much better.
Methods of an object gets or sets things about itself, not about others. You ask yourself if you can remove another? This is not intuitive.

I think you better use `match` or overwrite `__le__` so you can ask: `a < b`.
The question of its being absent or not is separated (you can use a property).
The caller then can say something like `if a.is_absent and a < b: ...`.

Reply:
`is_absent` is not intuitive for me since it is more a command to make several other routes absent and does not say that a certain route entry is absent. I was wondering if it should be renamed to `removes()` or so but was not so sure as well.

Reply:
Not sure what you mean. This object represents a route entry, which can be partial (in terms of its properties). How the object is used should not influence it's methods names.
`is_absent` is just a way to reflect what is its status, but it can be `is_state_absent` if it sounds better. The fact that makes it problematic to you (I am just guessing), is that this object is not representative of an actual route (due to the wildcard). You could argue that this is not a good representation, and that we need two object types, one that represents a real route and one that represents a "wildcard" route.